src/frontend/src/icp-eth/components/fee/FeeAmountDisplay.svelte

```diff
@@ -0,0 +1,50 @@
+<script lang="ts">
+	import { nonNullish } from '@dfinity/utils';
+	import { fade, slide } from 'svelte/transition';
+	import { formatToken } from '$lib/utils/format.utils';
+	import { EIGHT_DECIMALS } from '$lib/constants/app.constants';
+	import { i18n } from '$lib/stores/i18n.store';
+	import { replacePlaceholders } from '$lib/utils/i18n.utils';
+	import { balancesStore } from '$lib/stores/balances.store.js';
+	import { BigNumber } from '@ethersproject/bignumber';
+	import type { TokenId } from '$lib/types/token';
+	import { onDestroy } from 'svelte';
+
+	export let fee: BigNumber;
+	export let feeSymbol: string;
+	export let feeTokenId: TokenId;
+
+	const balance = nonNullish($balancesStore)
```

Comment:
Still not really convinced by that timeout I have to say.

Delay the animation still does not work?
If not, what about using debounce?

Reply:
It does not work because there is an issue in the code. `balance` should be reactive and not static.

```
// Not good
const balance = nonNullish($balancesStore) ...

// Better
let balance = nonNullish($balancesStore) ...
```

That said, fixing the issue fixes the animation but, it's a bit glitchy on first load. So here's your code:

```
<script lang="ts">
	import { debounce, nonNullish } from '@dfinity/utils';
	import { fade, slide } from 'svelte/transition';
	import { formatToken } from '$lib/utils/format.utils';
	import { EIGHT_DECIMALS } from '$lib/constants/app.constants';
	import { i18n } from '$lib/stores/i18n.store';
	import { replacePlaceholders } from '$lib/utils/i18n.utils';
	import { balancesStore } from '$lib/stores/balances.store.js';
	import { BigNumber } from '@ethersproject/bignumber';
	import type { TokenId } from '$lib/types/token';

	export let fee: BigNumber;
	export let feeSymbol: string;
	export let feeTokenId: TokenId;

	let balance: BigNumber | undefined;
	$: balance = nonNullish($balancesStore)
		? $balancesStore[feeTokenId]?.data ?? BigNumber.from(0n)
		: undefined;

	let insufficientFeeFunds = false;
	const debounceCheckFeeFunds = debounce(
		() => (insufficientFeeFunds = nonNullish(balance) && balance.lt(fee))
	);

	$: balance, debounceCheckFeeFunds();
</script>

<div transition:fade>
	{formatToken({
		value: fee,
		displayDecimals: EIGHT_DECIMALS
	})}
	{feeSymbol}
</div>
{#if insufficientFeeFunds && nonNullish(balance)}
	<p in:slide={{ duration: 250 }} class="text-cyclamen">
		{replacePlaceholders($i18n.send.assertion.not_enough_tokens_for_gas, {
			$balance: formatToken({
				value: balance,
				displayDecimals: EIGHT_DECIMALS
			}),
			$symbol: feeSymbol ?? ''
		})}
	</p>
{/if}
```

- balance should be reactive
- debounce should be a const to not be reactive
- `in:slide` instead of `transition:slide` is more pleasant

Reply:
one question: isn't this

```typescript
const debounceCheckFeeFunds = debounce(
  () => (insufficientFeeFunds = nonNullish(balance) && balance.lt(fee))
);

$: balance, debounceCheckFeeFunds();
```
	
	the same as this?
	
```typescript
$: {
  debounce(() => (insufficientFeeFunds = nonNullish(balance) && balance.lt(fee)), 250)();
}
```

in sense of reactivity?

and shouldn't we add `fee` as a reactive component of the debounce? so that if the fee updates, it checks again

Reply:
Not absolutely entirely sure, but from experience, I noticed the first method to be reliable while the second is not always.

The reason is that in the first method, you have one instance of the debounce that debounces a particular function. While in the second method, because of the reactivity, you might end up with multiple instances of the debounce and the function, and therefore the debounce itself does not work properly.

It might not be the case here, but as I said, from experience, I know the first method is reliable.

Does that make sense?

Reply:
Yes, understood, makes sense then to use the reliable one. And what about `fee` then? Shouldn't the code be:

`$: balance, fee, debounceCheckFeeFunds();`

Reply:
Yes, absolutely. Good catch.